pulp_ansible/tests/upgrade/pre/test_collection_install.py

```diff
@@ -0,0 +1,77 @@
+"""Tests that Collections hosted by Pulp can be installed by ansible-galaxy."""
+from os import path
+import subprocess
+import tempfile
+import unittest
+
+from pulpcore.client.pulp_ansible import (
+    DistributionsAnsibleApi,
+    RepositoriesAnsibleApi,
+    AnsibleRepositorySyncURL,
+    RemotesCollectionApi,
+)
+from pulp_smash.pulp3.bindings import monitor_task
+from pulp_smash.pulp3.utils import gen_distribution, gen_repo
+
+from pulp_ansible.tests.functional.constants import (
+    ANSIBLE_DEMO_COLLECTION,
+    ANSIBLE_DEMO_COLLECTION_REQUIREMENTS as DEMO_REQUIREMENTS,
+    GALAXY_ANSIBLE_BASE_URL,
+)
+from pulp_ansible.tests.functional.utils import gen_ansible_client, gen_ansible_remote
+from pulp_ansible.tests.functional.utils import set_up_module as setUpModule  # noqa:F401
+
+
+class InstallCollectionTestCase(unittest.TestCase):
+    """Test whether ansible-galaxy can install a Collection hosted by Pulp."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.client = gen_ansible_client()
+        cls.repo_api = RepositoriesAnsibleApi(cls.client)
+        cls.remote_collection_api = RemotesCollectionApi(cls.client)
+        cls.distributions_api = DistributionsAnsibleApi(cls.client)
+
+    def create_install_scenario(self, body, collection_name):
+        """Create Install scenario."""
+        repo = self.repo_api.create(gen_repo(name="pre_upgrade"))
```

Comment:
This repo isn't cleaned up in the post tests. Probably not a problem if this is only ran in the CI.

Reply:
Nice catch!